includes/class-dispatcher.php

```diff
@@ -125,25 +125,35 @@ public function _ajax_dispatch() {
 
     public function get_view_locations() {
         $dirs = array();
-        $dirs[] = WPBDP_PATH . 'includes/views/';
+        $dirs[] = WPBDP_PATH . 'includes/controllers/pages/';
         $dirs[] = WPBDP_PATH . 'core/views/';
 
         return apply_filters( 'wpbdp_view_locations', $dirs );
     }
 
     public function load_view( $view_name, $args = null ) {
+        $class_view_name = str_replace( '_', '-', $view_name );
         // TODO: add some filters so plugins can override default view loading.
         $filenames = array( $view_name . '.php',
-                            'views-' . $view_name . '.php' );
-
+                            'views-' . $view_name . '.php',
+                            'class-' . $class_view_name . '.php',
+                        );
+        $find = array( '.php', 'class' );
+        $replace = array( '', '' );
         foreach ( $this->get_view_locations() as $dir ) {
             foreach ( $filenames as $f ) {
                 $path = wp_normalize_path( WPBDP_FS::join( $dir, $f ) );
 
                 if ( ! file_exists( $path ) )
                     continue;
 
-                $classname = 'WPBDP__Views__' . implode( '_', array_map( 'ucfirst', explode( '_', str_replace( '.php', '', $f ) ) ) );
+                // Views that start with class- in the file name. Addons may not have this hence the condition.
+                if ( substr( $f, 0, 5 ) === 'class' ) {
+                    $f = str_replace( '-', '_', $f );
+                    $classname = 'WPBDP__Views_' . implode( '_', array_map( 'ucfirst', explode( '_', str_replace( $find, $replace, $f ) ) ) );
+                } else {
+                    $classname = 'WPBDP__Views__' . implode( '_', array_map( 'ucfirst', explode( '_', str_replace( '.php', '', $f ) ) ) );
```

Comment:
It looks like we can prevent this line twice like this:

Set $replace = ''; higher instead of $replace = array( '', '' );

```
$find = array( '.php', 'class' );
} else {
$find = '.php';
}
$classname = 'WPBDP__Views_' . implode( '_', array_map( 'ucfirst', explode( '_', str_replace( $find, $replace, $f ) ) ) );
```

Reply:
There is a difference in the two lines. With the `class` introduction the name would have an extra underscore. Hence the `if/else` is `WPBDP__Views_` and `WPBDP__Views__`

Reply:
Where does the extra underscore come from?

Reply:
We only remove the `class` prefix and it leaves an extra `-` that is replaced with an extra underscore. Also the file check if different in the updated code

Reply:
So can we use 'class-' for replace so we can remove that extra line?

Reply:
Other addons load their own views using the class, for now we have to keep both conditions in place

Reply:
Let's not mark conversations resolved until it's done.

This is brand new, so it wouldn't be used anywhere with 'class' in the filename yet. Right? Can you give me an example of a time where using 'class-' for the find wouldn't work?

Reply:
I've made a few changes to check for `class-` and it works both for [claim-listings](https://github.com/Strategy11/business-directory-claim-listings/blob/master/views/confirm_listing_claim.php) and the new file structure. I've pushed cleaner code for the checks

Reply:
This looks much better now. Thank you for sticking it out!

Reply:
A great learning experience :) Thank you for the guidance